returnn/tf/layers/basic.py

```diff
@@ -4764,13 +4764,29 @@ def __init__(self, eval, **kwargs):
 
 class CompareLayer(LayerBase):
   """
-  Compares (e.g. equality check) all the sources element-wise.
+  Compares element-wise the tokens of all input sequences among themselves and/or with a specified given value.
+  The comparisons are performed in a chain according to the order in which they are listed.
+  The final result is the logical "and" of all comparisons.
+
+  Example:
+    {"class": "compare", "from": ["i1", "i2"], "value": val, "kind": "less"}
+    computes i1 < i2 < val and it is true only if the whole chain of operations is true.
+
+  This layer is used in particular as the `end` node in an `output` subnetwork to specify the stopping criterion for generation, 
```

Comment:
Note: This is only a common example where it is used. But:
* The `end` layer doesn't need to be a `CompareLayer`. It could be anything else which determines the end.
* This is not the only use of the `CompareLayer`. It is always useful when you want to compare things.

Reply:
This comment is mostly about the formulation. I would maybe just reformulate it, like:
```suggestion
  A common example usage is the `end` layer in a rec subnetwork to specify the stopping criterion,
```